<launch>

  <!-- Load joint controller configurations from YAML file to parameter server -->
  <rosparam file="$(find furbot_control)/config/furbot_control.yaml" command="load"/>

  <!-- load the controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="--namespace=/furbot joint_state_controller rightRearWheel_velocity_controller leftRearWheel_velocity_controller rightFrontWheel_position_controller leftFrontWheel_position_controller"/>

  <!-- convert joint states to TF transforms for rviz, etc -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen">
    <param name="robot_description" command="$(find xacro)/xacro.py '$(find furbot_description)/urdf/furbot.xacro'" />
    <remap from="/joint_states" to="/furbot/joint_states" />
  </node>

  <!-- TOPICS AGGREGATION -->
  <node pkg="furbot_control" type="aggregate_topics" name="aggregate_topics" output="log"/>

  <!-- TELEOP -->
  <rosparam param="joy_node/dev">/dev/input/js1</rosparam>

  <node pkg="turtlesim" type="turtle_teleop_key" name="turtle_teleop_key" output="log" />
  <node pkg="furbot_control" type="teleop_furbot" name="teleop_furbot" output="log" />
  <node pkg="joy" type="joy_node" name="joy_node" output="screen" />

  <!-- ODOMETRY -->
  <node pkg="furbot_control" type="furbot_odometry" name="furbot_odometry" output="log" />


</launch>